docs/HowSwiftImportsCModules.md

```diff
@@ -0,0 +1,922 @@
+either platform.
+
+Nevertheless, mapping C `long` to `Int` does not work universally.
+Specifically, it does not work on LLP64 platforms (for example, Windows
+x86\_64), where C `long` is 32-bit and Swift's `Int` is 64-bit. On LLP64
+platforms, C `long` is mapped to Swift's explicitly-sized `Int32`.
+
+Typedefs for integer types in the C standard library are mapped like this (from
+[`swift.git/swift/lib/ClangImporter/MappedTypes.def`](../lib/ClangImporter/MappedTypes.def):
+
+| C and C++ types  | Swift types |
+| ---------------- | ----------- |
+| `uint8_t`        | `UInt8`     |
+| `uint16_t`       | `UInt16`    |
+| `uint32_t`       | `UInt32`    |
+| `uint64_t`       | `UInt64`    |
+| `int8_t`         | `Int8`      |
+| `int16_t`        | `Int16`     |
+| `int32_t`        | `Int32`     |
+| `int64_t`        | `Int64`     |
+| `intptr_t`       | `Int`       |
+| `uintptr_t`      | `UInt`      |
+| `ptrdiff_t`      | `Int`       |
+| `size_t`         | `Int`       |
+| `rsize_t`        | `Int`       |
+| `ssize_t`        | `Int`       |
+
+```c
+// C header.
+
+double Add(int x, long y);
+```
+
+```swift
+// C header imported in Swift.
+
+func Add(_ x: CInt, _ y: CLong) -> CDouble
+```
+
+# Free functions
+
+C functions are imported as free functions in Swift. Each type in the signature
+of the C function is mapped to the corresponding Swift type.
+
+## Argument labels
+
+Imported C functions don't have argument labels in Swift by default.  Argument
+labels can be added by API owners through annotations in the C header.
+
+```c
+// C header.
+
+#define SWIFT_NAME(X) __attribute__((swift_name(#X)))
+
+// No argument labels by default.
+void drawString(const char *, int xPos, int yPos);
+
+// The attribute specifies the argument labels.
+void drawStringRenamed(const char *, int xPos, int yPos)
+    SWIFT_NAME(drawStringRenamed(_:x:y:));
+```
+
+```swift
+// C header imported in Swift.
+
+func drawString(_: UnsafePointer<CChar>!, _ xPos: Int, _ yPos: Int)
+func drawStringRenamed(_: UnsafePointer<CChar>!, x: Int, y: Int)
+
+drawString("hello", 10, 20)
+drawStringRenamed("hello", x: 10, y: 20)
+```
+
+## Variadic arguments
+
+As of Swift 5, C functions with variadic arguments are not imported into Swift,
+however, there are no technical reasons why they can't be imported.
+
+Note that functions with `va_list` arguments are imported into Swift. `va_list`
+corresponds to `CVaListPointer` in Swift.
+
+C APIs don't define a lot of variadic functions, so this limitation has not
+caused a big problem so far.
+
+Often, for each variadic function there is a corresponding function that takes a
+`va_list` which can be called from Swift. A motivated developer can write an
+overlay that exposes a Swift variadic function that looks just like the C
+variadic function, and implement it in terms of the `va_list`-based C API. See
+usages of the `withVaList` function in the [`swift.git/stdlib`](`../stdlib`)
```

Comment:
Broken link:

```suggestion
usages of the `withVaList` function in the [`swift.git/stdlib`](../stdlib)
```

[Using Imported C Functions in Swift](https://developer.apple.com/documentation/swift/imported_c_and_objective-c_apis/using_imported_c_functions_in_swift#2992073) contains a `withVaList` example.

Reply:
Fixed the link, and added a link to Apple's docs.